src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -133,6 +136,30 @@ public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
 
     internal ITestRequestManager? TestRequestManager { get; set; }
 
+    private void UpdateDotnetLookupPath()
+    {
+        const string dotnetRootOverrideEnvVarName = "VSTEST_WINAPPHOST_DOTNET_ROOT";
+        const string programFilesPathEnvVarName = "ProgramFiles";
+        const string dotnetGlobalInstallationDir = "dotnet";
+
+        // Allow users to specify a dotnet private installation path before attempting to use the
+        // global installation path.
+        if (_environmentVariableHelper.GetEnvironmentVariable(dotnetRootOverrideEnvVarName) is not null)
+        {
+            return;
+        }
+
+        var dotnetPathPrefix = _environmentVariableHelper.GetEnvironmentVariable(programFilesPathEnvVarName);
+        if (dotnetPathPrefix is null)
+        {
+            return;
+        }
+
+        _environmentVariableHelper.SetEnvironmentVariable(
```

Comment:
Looks to me that we're changing this env var for "current" process and so we're affecting not only our host, is it right?
Should it be done before the host start?

Reply:
Indeed, we altered the environment of the current process. I've removed this fix and instead propagated the env variables only to the testhost process.